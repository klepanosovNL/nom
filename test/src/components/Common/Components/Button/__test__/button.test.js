import React from 'react';
import renderer from 'react-test-renderer';
import { Button } from '../Button';

describe('Button"s tests', () => {
	it('Sent only className', () => {
		const tree = renderer
			.create(<Button className='example'></Button>)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('Sent only children', () => {
		const tree = renderer.create(<Button>Hello</Button>).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('Sent all arguments', () => {
		const tree = renderer
			.create(
				<Button className='example' onClick={() => {}}>
					Hello
				</Button>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('Sent nothing', () => {
		const tree = renderer.create(<Button></Button>).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
