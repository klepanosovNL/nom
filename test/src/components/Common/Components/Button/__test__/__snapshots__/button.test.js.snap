// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button"s tests Sent all arguments 1`] = `
<button
  className="btn_default example"
  onClick={[Function]}
>
  Hello
</button>
`;

exports[`Button"s tests Sent nothing 1`] = `
<button
  className="btn_default "
/>
`;

exports[`Button"s tests Sent only children 1`] = `
<button
  className="btn_default "
>
  Hello
</button>
`;

exports[`Button"s tests Sent only className 1`] = `
<button
  className="btn_default example"
/>
`;
